src/js/protocols/stm32usbdfu.js

```diff
@@ -775,6 +775,7 @@ STM32DFU_protocol.prototype.upload_procedure = function (step) {
                     for (var j = 0; j < self.flash_layout.sectors[i].num_pages; j++) {
                         if (self.options.erase_chip) {
                             // full chip erase
+                            $('span.hardwareId').hide();
```

Comment:
I think the data driven approach is better - we want to reset the data (with `FC.resetState()`) when it becomes invalid (disconnect of the device), and not 'combat the symptoms' by hiding the element that displays cached invalid data.

Reply:
@mikeller Yes, it makes sense, I wasn't really sure which was the best option, done!